base/inference.jl

```diff
@@ -3657,7 +3657,7 @@ function inlineable(f::ANY, ft::ANY, e::Expr, atypes::Vector{Any}, sv::Inference
     if sv.params.inlining
         if isa(e.typ, Const) # || isconstType(e.typ)
             if (f === apply_type || f === fieldtype || f === typeof ||
-                istopfunction(topmod, f, :typejoin) ||
+                istopfunction(topmod, f, :typejoin) || f === (===) ||
```

Comment:
note that `===` should often show up here as a `Conditional` object, not a `Const`